restapi/src/main/java/io/stargate/web/docsapi/service/query/search/resolver/filter/impl/InMemoryCandidatesFilter.java

```diff
@@ -0,0 +1,139 @@
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * {{@link CandidatesFilter} that works with set of {@link FilterExpression}s that contain only
+ * in-memory conditions.
+ */
+public class InMemoryCandidatesFilter implements CandidatesFilter {
+
+  private final Collection<FilterExpression> expressions;
+
+  private final FilterPathSearchQueryBuilder queryBuilder;
+
+  private final ExecutionContext context;
+
+  private InMemoryCandidatesFilter(
+      Collection<FilterExpression> expressions, ExecutionContext context) {
+    boolean hasPersistence =
+        expressions.stream().anyMatch(e -> e.getCondition().isPersistenceCondition());
+
+    if (hasPersistence) {
+      throw new IllegalArgumentException(
+          "InMemoryCandidatesDocumentsResolver works only with the non persistence conditions.");
+    }
+
+    this.expressions = expressions;
+    this.queryBuilder = new DocumentSearchQueryBuilder(expressions);
+    this.context = createContext(context, expressions);
+  }
+
+  public static Function<ExecutionContext, CandidatesFilter> forExpression(
+      FilterExpression expression) {
+    return forExpressions(Collections.singletonList(expression));
+  }
+
+  public static Function<ExecutionContext, CandidatesFilter> forExpressions(
+      Collection<FilterExpression> expressions) {
+    return context -> new InMemoryCandidatesFilter(expressions, context);
+  }
+
+  @Override
+  public Single<? extends Query<? extends BoundQuery>> prepareQuery(
+      DataStore dataStore, DocsApiConfiguration configuration, String keyspace, String collection) {
+    FilterPath filterPath = queryBuilder.getFilterPath();
+    // resolve depth we need
+    String[] neededColumns =
+        QueryConstants.ALL_COLUMNS_NAMES.apply(filterPath.getPath().size() + 1);
+    // we can only fetch one row if path is fixed
+    Integer limit = filterPath.isFixed() ? 1 : null;
+    return RxUtils.singleFromFuture(
+            () -> {
+              BuiltQuery<? extends BoundQuery> query =
+                  queryBuilder.buildQuery(
+                      dataStore::queryBuilder, keyspace, collection, limit, neededColumns);
+              return dataStore.prepare(query);
+            })
+        .cache();
+  }
+
+  @Override
+  public Maybe<RawDocument> bindAndFilter(
+      QueryExecutor queryExecutor,
+      DocsApiConfiguration configuration,
+      Query<? extends BoundQuery> preparedQuery,
+      RawDocument document) {
+    BoundQuery query = preparedQuery.bind(document.id());
+
+    // query, take one, test against expression
+    FilterPath filterPath = queryBuilder.getFilterPath();
+
+    // page size 2 with limit 1 to ensure no extra page fetching (only on fixed path)
+    int pageSize = filterPath.isFixed() ? 2 : configuration.getSearchPageSize();
+    return queryExecutor
+        .queryDocs(query, pageSize, null, context)
+        .take(1)
+        .filter(matchAll(expressions))
+        .singleElement();
```

Comment:
What about missing properties? If we have `where missing NE "value"` the code on `master` apparently treats it as `null NE "value"` and will accept the doc. Current logic will not select any rows and produce no output. WDYT?

Reply:
hmm, this should not be the case, double checking..

Reply:
damn, this is an issue, I am not loading the proper "rows" for the `NE` and `NIN`, as they filter path non-matching should also be included here.. Problem is that old solution was loading all the rows for the in-memory filtering and I tried to avoid this.. For candidates filtering this is really not an issue and could be solved easily, but for the memory doc fetching it is a problem and we might need to go back to fetching all here..

Reply:
Fixed.